packages/remix-dev/compiler/plugins/cssFilePlugin.ts

```diff
@@ -0,0 +1,96 @@
+import * as path from "path";
+import * as fse from "fs-extra";
+import esbuild from "esbuild";
+
+import { BuildMode } from "../../build";
+import type { BuildConfig } from "../../compiler";
+
+/**
+ * This plugin loads css files with the "css" loader (bundles and moves assets to assets directory)
+ * and exports the url of the css file as its default export.
+ */
+export function cssFilePlugin(
+  buildConfig: Partial<BuildConfig>
```

Comment:
Since we're only using `buildConfig`'s `mode`, this can be

```suggestion
  buildConfig: Pick<BuildConfig, "mode">
```

Reply:
Improved that in [362250b](https://github.com/remix-run/remix/pull/4130/commits/362250bcad6b7a82980c3498ed9c88ed9b9bec03)